js_modules/dagster-ui/packages/ui-core/src/asset-graph/ContextMenuWrapper.tsx

```diff
@@ -18,14 +18,24 @@ export const ContextMenuWrapper = ({
   wrapperInnerStyles?: React.CSSProperties;
 }) => {
   const [menuVisible, setMenuVisible] = React.useState(false);
-  const [menuPosition, setMenuPosition] = React.useState<{top: number; left: number}>({
-    top: 0,
-    left: 0,
+  const [menuPosition, setMenuPosition] = React.useState<{
+    x: number;
+    y: number;
+    anchor: 'left' | 'right';
+  }>({
+    anchor: 'left',
+    x: 0,
+    y: 0,
   });
 
   const showMenu = (e: React.MouseEvent) => {
+    const anchor = window.innerWidth - e.pageX < 240 ? 'right' : 'left';
```

Comment:
instead of hardcoding 240 here should we take it as a prop with maybe 240 as the default? We could also use `useViewport` to get the real width I think?

Reply:
edit hrm, it looks like we have to render hidden, size, render visible to do this with the correct breakpoint, going to come back to it in a follow-up!